Automattic/jetpack
# Decouple Publicize connection list access from UI scripts

Fixes #9039 (along with PR #9144)

**Note**: These changes are standalone but they are a dependency of #9144. They were originally contained in #9144, but are being pulled out for for clarity/organization because that PR was getting too large and complex.

#### Changes proposed in this Pull Request:
This PR separates the Publicize connection list processing from the HTML generation for the UI form.

##### Before this PR:
The method `Publicize_UI::get_metabox_form_connected()` in `ui.php` retrieved and filtered the connection list and then also generated the HTML form. 

##### After this PR:
This PR leaves the HTML generation in `Publicize_UI::get_metabox_form_connected()` but moves the connection gathering/filtering to new method `Publicize::get_filtered_connection_data()` in `publicize-jetpack.php`.

##### Why this is needed:
PR #9144 needs to access the connection list so it can be JSON encoded and utilized in new React-based frontend for Gutenberg. Splitting the connection list retrieval to a separate method allows code reuse between the classic editor HTML generation and the Gutenberg front-end to gather.

The final diff view of these changes looks a little muddled, but the [first commit](https://github.com/Automattic/jetpack/pull/9466/commits/6658192891d80a2f120304f80c5341efe7ff1f1a) pretty effectively shows the basics of what's being moved where in this PR.

This PR should not affect the Classic Editor in any way.

Unit tests have been added to test connection data fields and to test the three potentially affected filters:
 - `'wpas_submit_post?'`
 - `'publicize_checkbox_global_default'`
 - `'publicize_checkbox_default'`

#### Testing instructions:
Checkout and test Publicize in Classic Editor to confirm there are no functional changes.
See #9144 for additional testing instructions.



props to @ehg for the design suggestions!
